gimmik/kernels/cuda/cstream-ksplit.mako

```diff
@@ -0,0 +1,81 @@
+<%inherit file='base'/>
+
+<%
+kparts = partition(A, ksplit, by='cols')
+cchunks = chunk(range(m), csz)
+loaded = set()
+%>
+
+__global__ void
+% if n is None:
+${kname}(int n,
+         const ${dtype}* __restrict__ b, int ldb,
+         ${dtype}* __restrict__ c, int ldc)
+{
+  % if width > 1:
+    n = ((n + ${width} - 1) / ${width}) * ${width};
+    ldb /= ${width};
+    ldc /= ${width};
+  % endif
+% else:
+${kname}(const ${dtype}* __restrict__ b, ${dtype}* __restrict__ c)
+{
+    const int n = ${-(-n // width)};
+    const int ldb = ${ldb // width};
+    const int ldc = ${ldc // width};
+% endif
+    const int i = blockDim.x*blockIdx.x + threadIdx.x;
+
+    ${dtype} cv[${-(-csz // ksplit)}], bv[${-(-k // ksplit)}], dotp;
+    __shared__ ${dtype} csub[${ksplit - 1}][${csz}][${blockx}];
+
+    if (i >= n)
+        return;
+
+## Iterate over the column-partitions of B
+% for bid, kbx in enumerate(kparts):
+    if (threadIdx.y == ${bid})
+    {
+  ## Iterate over the row-partitions of C
+  % for cchunk in cchunks:
+    ## Evaluate our partial dot products
+    % for j in cchunk:
+      ## Load in any missing parts of B
+      % for kx in kbx:
+        % if A[j, kx] != 0 and kx not in loaded:
+        bv[${loop.index}] = __ldcg(b + i + ${kx}*ldb); <% loaded.add(kx) %>
```

Comment:
Are ldcg betting in all the cases you've tested? Same point as above.

Reply:
Inspected the generated SASS and it is indeed different to the compiler's default.  Prevents a line from being allocated in L1 which can be useful if the kernel is spilling to local memory as it reduces the chance of an eviction.